### PR TITLE
chore: Add nightly build workflow

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add nightly build workflow.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        GOLINT=`go list -f {{.Target}} golang.org/x/lint/golint`
+        $GOLINT -set_exit_status ./...
+
+    - name: Run Tests
+      run: ./.github/scripts/run_all_tests.sh
+      env:
+        FIREBASE_SERVICE_ACCT_KEY: ${{ secrets.FIREBASE_SERVICE_ACCT_KEY }}
+        FIREBASE_API_KEY: ${{ secrets.FIREBASE_API_KEY }}
+
+    - name: Send email on failure
```
Comment (hiranya911) [inline .github/workflows/nightly.yml:55]: I've been thinking if we should try and reduce the duplication here. Something like:

```
- name: Generate email text
  if: failure() || cancelled()
  run: ./.github/scripts/generate_email.sh
  env:
    WORKFLOW_CANCELLED: ${{ cancelled() }}
    WORKFLOW_FAILED: ${{ failure() }}

- name: Send email
  uses: firebase/firebase-admin-node/.github/actions/send-email@master
  ...
```

No need to do this urgently. But something to think about.

**Review by hiranya911 — APPROVED** [review]
Thanks. LGTM with a suggestion for future improvement.

**Comment by hiranya911** [comment]
I've changed base branch to dev.

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (lahirumaramba) [reply .github/workflows/nightly.yml:55]: Good call!
I will use one of the other repos to test this out first. Thanks!

**Comment by lahirumaramba** [comment]
> I've changed base branch to dev.

Thank you!!

**Comment by lahirumaramba** [comment]
@googlebot I fixed it.

**Pull request merged**